plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -858,23 +939,117 @@ SecRule REQUEST_FILENAME "@contains /apps/notes/notes/" \
+SecRule REQUEST_FILENAME "@endsWith /apps/notes/settings" \
+    "id:9508921,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
 #
 # [ Custom CSS ]
 #
 
 # Allow admin to add custom CSS code
 SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/provisioning_api/api/v[0-9]/config/apps/theming_customcss/customcss$" \
-   "id:9508930,\
+    "id:9508930,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:value,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0'"
+
+#
+# [ Nextcloud Circles ]
+#
+
+# When using Nextcloud circles
+# PUT - When editing/creating a circle
+# DELETE - When deleting an existing circle
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/" \
+    "id:9508940,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT DELETE'"
+
+# When setting a unique password for all shares to a circle
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/circles/circles/.*/setting$" \
+    "id:9508941,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.value,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0'"
+
+#
+# [ Nextcloud Photos ]
+#
+
+# When changing settings for Nextcloud photos
+SecRule REQUEST_FILENAME "@rx /apps/photos/api/v[0-9]/config/config/(?:croppedLayout|photosLocation)$" \
+    "id:9508950,\
     phase:1,\
     pass,\
     t:none,\
     nolog,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:value"    
-   
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
+#
+# [ Nextcloud Files ]
+#
+
+# Expanding share menus under Files --> Shares
+SecRule REQUEST_FILENAME "@rx /apps/files/api/v[0-9]/views/shareoverview/expanded$" \
+    "id:9508960,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
+# Changing settings within Nextcloud Files
+SecRule REQUEST_FILENAME "@rx /apps/files/api/v[0-9]/config/(?:show_hidden|crop_image_previews)$" \
+    "id:9508961,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
+# Changing sorting direction in trashbin
+SecRule REQUEST_FILENAME "@rx /apps/files/api/v[0-9]/views/trashbin/(?:sorting_direction|sorting_mode)$" \
+    "id:9508962,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
+
+# Rejecting a file ownership transfer
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/files/api/v[0-9]/transferownership/[0-9]*$" \
```

Comment:
Is there `DELETE` case when the last path segment is not a number?

Reply:
No, it's needed to distinguish between multiple file ownership transfer requests.